rust/src/modularity.rs

```diff
@@ -0,0 +1,129 @@
+//! Implementation of the client-side of "rpm-ostree module".
+
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::{anyhow, bail, Result};
+use gio::DBusProxyExt;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "rpm-ostree module", no_version)]
+#[structopt(rename_all = "kebab-case")]
+enum Opt {
+    /// Enable a module
+    Enable(InstallOpts),
+    /// Disable a module
+    Disable(InstallOpts),
+    /// Install a module
+    Install(InstallOpts),
+    /// Uninstall a module
+    Uninstall(InstallOpts),
+}
+
+#[derive(Debug, StructOpt)]
+struct InstallOpts {
+    #[structopt(parse(from_str))]
+    modules: Vec<String>,
+    #[structopt(long)]
+    reboot: bool,
+    #[structopt(long)]
+    lock_finalization: bool,
+    #[structopt(long)]
+    dry_run: bool,
+}
+
+const OPT_KEY_ENABLE_MODULES: &str = "enable-modules";
+const OPT_KEY_DISABLE_MODULES: &str = "disable-modules";
+const OPT_KEY_INSTALL_MODULES: &str = "install-modules";
+const OPT_KEY_UNINSTALL_MODULES: &str = "uninstall-modules";
+
+pub fn entrypoint(args: &[&str]) -> Result<()> {
+    let opt = Opt::from_iter(args.iter().skip(1));
+    match opt {
+        Opt::Enable(ref opts) => enable(opts)?,
+        Opt::Disable(ref opts) => disable(opts)?,
+        Opt::Install(ref opts) => install(opts)?,
+        Opt::Uninstall(ref opts) => uninstall(opts)?,
+    };
+    Ok(())
```

Comment:
(Super minor but this could be folded to `match Opt::from_iter()`, i.e. drop the `?` from each invocation and pass through the `Result` and drop the trailing `Ok(())`)

Reply:
Thanks, that is indeed cleaner!